core/src/rpc/verifier.rs

```diff
@@ -241,12 +230,7 @@ impl ClementineVerifier for Verifier {
 
         tracing::info!("Received deposit sign request");
 
-        let verifier_idx = self.idx;
-        let nofn_xonly_pk = self.nofn_xonly_pk;
-        let sessions = Arc::clone(&self.nonces);
-        let signer = self.signer.clone();
-        let verifiers_public_keys = self.config.verifiers_public_keys.clone();
-        let db_clone = self.db.clone();
+        let verifier = self.clone();
```

Comment:
Does this use Arc::clone inside?
I am not sure if this self.clone is cloning the whole sessions data which is not we want

Reply:
https://doc.rust-lang.org/std/sync/struct.Arc.html#cloning-references apparently it does run `Arc::Clone` for the `Arc` fields.

Reply:
Great!